spec/services/shelving_service_spec.rb

```diff
@@ -118,4 +138,261 @@
+    let(:shelving_service) { described_class.new(cocina_object) }
+
+    context 'with all subset' do
+      let(:previous_content_metadata) { read_fixture('content_diff_reports/jq937jp0017/v0001/metadata/contentMetadata.xml') }
+
+      # mock the current object content metadata (which would be converted from cocina for SDR object)
+      before do
+        allow(shelving_service)
+          .to receive(:content_metadata)
+          .and_return(read_fixture('content_diff_reports/jq937jp0017/v0002/metadata/contentMetadata.xml'))
+      end
+
+      it 'returns expected Moab::FileInventoryDifference' do
+        diff = shelving_service.send(:content_diff, 'all')
+        diff_ng_xml = Nokogiri::XML(diff.to_xml)
+        exp_xml = <<-XML
+          <fileGroupDifference groupId="content" differenceCount="3" identical="3" copyadded="0" copydeleted="0" renamed="0" modified="1" deleted="2" added="0">
+            <subset change="identical" count="3">
+              <file change="identical" basisPath="title.jpg" otherPath="same">
+                <fileSignature size="40873" md5="1a726cd7963bd6d3ceb10a8c353ec166" sha1="583220e0572640abcd3ddd97393d224e8053a6ad" sha256=""/>
+              </file>
+              <file change="identical" basisPath="page-2.jpg" otherPath="same">
+                <fileSignature size="39450" md5="82fc107c88446a3119a51a8663d1e955" sha1="d0857baa307a2e9efff42467b5abd4e1cf40fcd5" sha256=""/>
+              </file>
+              <file change="identical" basisPath="page-3.jpg" otherPath="same">
+                <fileSignature size="19125" md5="a5099878de7e2e064432d6df44ca8827" sha1="c0ccac433cf02a6cee89c14f9ba6072a184447a2" sha256=""/>
+              </file>
+            </subset>
+            <subset change="copyadded" count="0"/>
+            <subset change="copydeleted" count="0"/>
+            <subset change="renamed" count="0"/>
+            <subset change="modified" count="1">
+              <file change="modified" basisPath="page-1.jpg" otherPath="same">
+                <fileSignature size="25153" md5="3dee12fb4f1c28351c7482b76ff76ae4" sha1="906c1314f3ab344563acbbbe2c7930f08429e35b" sha256=""/>
+                <fileSignature size="32915" md5="c1c34634e2f18a354cd3e3e1574c3194" sha1="0616a0bd7927328c364b2ea0b4a79c507ce915ed" sha256=""/>
+              </file>
+            </subset>
+            <subset change="deleted" count="2">
+              <file change="deleted" basisPath="intro-1.jpg" otherPath="">
+                <fileSignature size="41981" md5="915c0305bf50c55143f1506295dc122c" sha1="60448956fbe069979fce6a6e55dba4ce1f915178" sha256=""/>
+              </file>
+              <file change="deleted" basisPath="intro-2.jpg" otherPath="">
+                <fileSignature size="39850" md5="77f1a4efdcea6a476505df9b9fba82a7" sha1="a49ae3f3771d99ceea13ec825c9c2b73fc1a9915" sha256=""/>
+              </file>
+            </subset>
+            <subset change="added" count="0"/>
+        </fileInventoryDifference>
```

Comment:
```suggestion
          </fileGroupDifference>
```

doesn't seem to make a difference in the test result (helpful nokogiri xml normalization making up for it?).  but i think this closing tag is slightly off from what it should be?

Reply:
Thanks - fixed this.